### PR TITLE
Index comments

### PR DESCRIPTION
**Pull request by gforcada**
Fixes:
https://github.com/collective/collective.solr/issues/96

### CODE REVIEW

File: src/collective/solr/tests/test_indexer.py [hunk 1]
```diff
@@ -33,6 +34,19 @@ def __call__(self):
         return True
 
 
+class Bar(CatalogAware):
```
Comment (tisto) [inline src/collective/solr/tests/test_indexer.py]: @gforcada can we rename Bar into something more suitable?

Comment (gforcada) [inline src/collective/solr/tests/test_indexer.py]: Sure, I used because some lines above we have ``"Foo"``

**Comment by gforcada** [comment]
@tisto renamed to ``Comentish``, sounds better?

**Comment by tisto** [comment]
@gforcada please rebase. Will merge afterwards.

**Comment by gforcada** [comment]
@tisto done

**Comment by gforcada** [comment]
@tisto (or anyone else) any change to get it merged? :-)

**Comment by gforcada** [comment]
**DO NOT MERGE** seems that some commits since I wrote this code (the atomic updates(?)) it no longer indexes comments again :-/ looking int it right now

**Comment by gforcada** [comment]
Well, *right now* was a bit exaggerated :-) Now, I do work on it.

**Comment by tisto** [comment]
@gforcada do you still plan to work on this? Otherwise I would close the pr...

**Comment by gforcada** [comment]
@tisto yes, I have it working for der Freitag, although the changes are not only here at collective.solr but also at plone.app.discussion *and* collective.indexing.

At which point at thought that, as the PLIP 1343 is nearly finished, instead of putting too much effort on bringing it separately on all these three packages, which one of them is still being major-refactored and ported to Plone 5, I would wait for that PLIP.

Then once c.solr is at Plone 5 I would work on bringing collective.solr to work on 5.1 as well.

Despite that if anyone wants to use it *now* the patches are in der Freitag forks:
- https://github.com/derFreitag/collective.solr
- https://github.com/derFreitag/plone.app.discussion
- https://github.com/derFreitag/collective.indexing

**Comment by tomgross** [comment]
What's the state of this?

**Comment by gforcada** [comment]
As said on my previous comments, there are some fixes to be pushed to p.a.discussion, collective.indexing and collective.solr itself, I wanted to wait until  the port to Plone 5 was done so that the pull requests were not tangling each other.

Should I push pull requests for each of them already?

**Comment by tisto** [comment]
@gforcada what's the status of this PR? Do you still plan to work on it?

**Comment by gforcada** [comment]
Yes, priorities did change, but this is still something that I want to get in. I will try a rebase to see if it works with the new solr version (nice work!! :wink: )

**Comment by gforcada** [comment]
@tisto finally green, please review :smile:

**Comment by gforcada** [comment]
Friendly ping to whoever might be interested in getting this merged :smile:

**Comment by gforcada** [comment]
@tisto sorry that you had to go through the hassle of merging it, but without the corresponding changes on p.a.discussion and collective.indexing (as it still depends on it) this changes probably do more harm than good :confused:

**Comment by gforcada** [comment]
Well, maybe there is no need to revert, but comments will not be indexed anyway. I will prepare the pull requests for p.a.discussion and c.indexing, let's see how they go...

**Comment by tisto** [comment]
@gforcada are you ok with reverting for now? It turns out this introduces a hard dependency on Archetypes (CatalogMultiplex) which breaks Plone 5.2.

**Comment by tisto** [comment]
@gforcada I reverted the PR merge. If you are ok with it I will make another release. If you want to continue to work on this, feel free to create a new PR.

**Comment by gforcada** [comment]
Continue to work on this? No thanks, for +3 years we, at der Freitag, already have a public fork of this repository. I will close the related pull requests then as well, as they are meaningless if this is not merged.

**Pull request merged**